app/src/main/java/com/mymeetings/pairpomodoro/view/MainActivity.kt

```diff
@@ -75,29 +82,36 @@ class MainActivity : AppCompatActivity(), ServiceConnection {
         startService(Intent(this, PomodoroService::class.java))
     }
 
-    private fun showStatus(pomodoroStatus: PomodoroStatus?, sharingKey: String?) {
+    private fun onStatusUpdated(pomodoroStatus: PomodoroStatus?) {
         if (pomodoroStatus != null) {
-            showTimerView(pomodoroStatus, sharingKey)
+            showTimerView(pomodoroStatus)
         } else {
             showSelectionView()
         }
     }
 
-    private fun keyNotFound() {
+    private fun onTimerCreated(
+        timerPreference: TimerPreference,
+        sharingKey: String
+    ) {
+        timerProgressBar.max = timerPreference.getFocusTime().toInt()
```

Comment:
The max should vary for different stages.
Eg. For focus -> getFocusTime
For ShortBreak -> the max should the getShortBreakTime

Reply:
Yeah correct, so we should have a type of timer to handle that?

Reply:
Yes. Timer gives PomodoroState in PomodoroStatus. We can use that.

Reply:
I added the logic. Please review.